packages/lib/src/vis/hooks.ts

```diff
@@ -104,3 +104,36 @@ export function useCustomColors(
 
   return [colors, refCallback];
 }
+
+export function useRaycasterThreshold(
+  lineOrPoints: 'Line' | 'Points',
+  desiredThreshold: number
+) {
+  const raycaster = useThree((state) => state.raycaster);
+  const zoomVector = useThree((state) => state.camera.scale);
+
+  useEffect(() => {
+    const params = raycaster.params[lineOrPoints];
+    const oldThreshold = params ? params.threshold : 1;
+
+    if (params) {
+      params.threshold = desiredThreshold;
+    }
+
+    return () => {
+      if (params) {
+        params.threshold = oldThreshold;
+      }
+    };
+  }, [raycaster, lineOrPoints, desiredThreshold]);
+
+  useFrame(() => {
+    // Reduce the threshold dynamically when zooming as the size of Line and Points objects is reduced in world units
```

Comment:
I feel like this comment could be a bit clearer. The lines and points keep the same size in world space -- they only change in camera space and HTML space, if I'm not mistaken.